packages/theme-check-common/src/JSONValidator.ts

```diff
@@ -25,6 +30,11 @@ export class JSONValidator {
     });
   }
 
+  public parse: ParseJSON<SourceCodeType> = (sourceCode, jsonString) => {
+    const jsonTextDocument = TextDocument.create('file:' + sourceCode.uri, 'json', 0, jsonString);
+    return this.service.parseJSONDocument(jsonTextDocument);
+  };
+
```

Comment:
I feel like we should use the same AST we use for JSON files 🤔 ?

https://github.com/Shopify/theme-tools/blob/main/packages/theme-check-common/src/to-source-code.ts#L16C1-L22C2

Reply:
I'm not 100% those two ASTs have the same shape and it would be weird if we had two different AST shapes for different checks. 

Also I believe that we'll want to do this in different checks, so this result should be cacheable so other checks can get the same AST without doing the parse multiple times.